plugin/trino-spooling-filesystem/src/main/java/io/trino/spooling/filesystem/FileSystemSpoolingManager.java

```diff
@@ -166,26 +163,35 @@ public SpooledSegmentHandle handle(SpooledLocation location)
             return FileSystemSpooledSegmentHandle.of(queryId, uuid, Optional.empty());
         }
 
-        long encryptionKeyHash = input.readLong();
-        List<String> encryptionCipher = location.headers().get(ENCRYPTION_KEY_CIPHER);
+        Slice key = getEncryptionKey(location.headers());
+        return FileSystemSpooledSegmentHandle.of(queryId, uuid, Optional.of(key));
+    }
+
+    private static Slice getEncryptionKey(Map<String, List<String>> headers)
+    {
+        List<String> encryptionCipher = headers.get(ENCRYPTION_KEY_CIPHER);
```

Comment:
Why are we not failing if List has more than 1 element?

Reply:
makes sense. I'll add it